### ISSUE TITLE
about validation and test images

### ISSUE THREAD

**Issue by nbbllxx0** [ayu-22/BPPNet-Back-Projected-Pyramid-Network#11]
I tried to train with Ohaze dataset, with batch len=210 (default setting), 35 images for training, 5 for test, i have 2 problems that need some help, first, which images are your test ones ?except 2 images shown on the gif i selected , which 3 left are your testing choices? second, except 5 for test 35 for training, there are still 5 left for validation, but i cant find any validation codes in file, should i put these 5 images in training? for i cant get that high psnr and ssim as paper declaimed, specifically, psnr 19.1+- with ssim 0.75+, thank you

**Comment by ayu-22** [comment]
@nbbllxx0 We have done testing on validation data because we didn't have ground truth for the test images. The scores mentioned in our paper are on validation data for O-HAZE. You can also refer to our paper for images. 
Regarding the results, have you tried to decrease the learning rate once the loss becomes stable? If you have not done this please do that.

**Comment by nbbllxx0** [comment]
after training with 300 epochs with lr=0.00005( 5times than defalut 0.00001) on Ohaze dataset, reaching mse: +0.0009544164370739214 | ssim: 0.949728613781138 | unet:0.8520574411509726 ,the mse is still slightly decreasing; tested on 10 images that excluded from 35 Ohaze training images, led to the result with psnr 22.19 & ssim 0.8667; much higher than before but still a longway to go from psnr 24.27 &ssim 0.8919 as the paper claims ; i guess the mse could be stable in at least 500(or more) epochs with default lr=0.00001, how many epochs did you run to achieve such high psnr with lr = 0.00001 on Ohaze? Thanks

**Comment by ayu-22** [comment]
We only tested on validation data which had only 5 images. As you are testing on 10 images (my guess is 5 of validation images and 5 of test images), the ssim and psnr can differ. Also, we started training with an LR of 0.001 (you can refer to the "training details" section of our paper), 0.00001 was the final LR. regarding the number of the epochs, we don't exactly remember it.

**Comment by nbbllxx0** [comment]
Thanks, that is all i need



---Original---
From: "Ayush ***@***.***&gt;
Date: Sat, Jun 5, 2021 14:56 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ayu-22/BPPNet-Back-Projected-Pyramid-Network] about validation and test images (#11)




 
We only tested on validation data which had only 5 images. As you are testing on 10 images (my guess is 5 of validation images and 5 of test images), the ssim and psnr can differ. Also, we started training with an LR of 0.001 (you can refer to the "training details" section of our paper), 0.00001 was the final LR. regarding the number of the epochs, we don't exactly remember it.

**Comment by nbbllxx0** [comment]
by mutiple times of training, reaching psnr 23.5&ssim 0.8807 , could you please offer your generator.pth &discriminator.pth file which achieving the finest psnr&ssim in your paper? thanks

**Comment by gina7152316** [comment]
I also need pretrained models, could auther please offer generator.pth &discriminator.pth files which achieving the finest psnr&ssim in your paper? 
thanks

**Comment by ayu-22** [comment]
@nbbllxx0 @gina7152316 Please sent a mail to my email id mention in the paper regarding weights, I will share the weights in that mail reply.

**Comment by gina7152316** [comment]
I just sent a mail which is mentioned in the paper. 
Looking forward to hearing from you.

**Comment by nbbllxx0** [comment]
Just found your reply which mentioned your pretrained model , I would be really grateful if you could share your weights of BPP training, thank you

---Original---
From: "Ayush ***@***.***&gt;
Date: Sun, Jul 18, 2021 22:00 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ayu-22/BPPNet-Back-Projected-Pyramid-Network] about validation and test images (#11)




 
@nbbllxx0 @gina7152316 Please sent a mail to my email id mention in the paper regarding weights, I will share the weights in that mail reply.

**Comment by ayu-22** [comment]
@nbbllxx0 I would be happy to provide you pre-trained weights. Please send a mail to my email id mentioned in the paper. I will share the weights in that mail reply.

**Comment by Beelzbube** [comment]
> Just found your reply which mentioned your pretrained model , I would be really grateful if you could share your weights of BPP training, thank you
> […](#)
> ---Original--- From: "Ayush ***@***.***&gt; Date: Sun, Jul 18, 2021 22:00 PM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [ayu-22/BPPNet-Back-Projected-Pyramid-Network] about validation and test images (#11) @nbbllxx0 @gina7152316 Please sent a mail to my email id mention in the paper regarding weights, I will share the weights in that mail reply. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

1466148452@qq.com
I would appreciate it if you could send the pre-trained weights to me. I'm looking forward to your early reply ovo.

**Comment by Beelzbube** [comment]
> @nbbllxx0 I would be happy to provide you pre-trained weights. Please send a mail to my email id mentioned in the paper. I will share the weights in that mail reply.

1466148452@qq.com
I would appreciate it if you could send the pre-trained weights to me. I'm looking forward to your early reply ovo.

**Comment by Beelzbube** [comment]
> I just sent a mail which is mentioned in the paper. Looking forward to hearing from you.

1466148452@qq.com
I would appreciate it if you could send the pre-trained weights to me. I'm looking forward to your early reply ovo.

**Comment by islandLZ** [comment]
2466494557@qq.com
I would appreciate it if you could send the pre-trained weights and the best weights to me. I'm looking forward to your early reply ovo.
